File: app_java/src/main/java/tech/thdev/android_mvp_sample/data/source/image/SampleImageLocalDataSource.java
package tech.thdev.android_mvp_sample.data.source.image;

import android.content.Context;

import java.util.ArrayList;

import tech.thdev.android_mvp_sample.data.ImageItem;

/**
 * Created by tae-hwan on 1/30/17.
 */

public class SampleImageLocalDataSource implements SampleImageSource {

    @Override
    public void getImages(Context context, int size, LoadImageCallback loadImageCallback) {
        ArrayList<ImageItem> items = new ArrayList<>();

        for (int i = 0; i < size; i++) {
            final int random = (int) (Math.random() * 15);
            final String name = String.format("sample_%02d", random);
            final int resource = context.getResources().getIdentifier(name, "drawable", context.getApplicationContext().getPackageName());
            items.add(new ImageItem(resource, name));
        }

        if (loadImageCallback != null) {
            loadImageCallback.onImageLoaded(items);
        }
    }
}

File: app_kotlin/src/main/java/tech/thdev/app_kotlin/data/source/image/SampleImageLocalDataSource.kt
package tech.thdev.app_kotlin.data.source.image

import android.content.Context
import tech.thdev.app_kotlin.data.ImageItem
import java.util.*

/**
 * Created by tae-hwan on 1/30/17.
 */
object SampleImageLocalDataSource : SampleImageSource {

    override fun getImages(context: Context, size: Int, loadImageCallback: SampleImageSource.LoadImageCallback?) {
        val list = ArrayList<ImageItem>()
        for (index in 0..size) {
            val name = String.format("sample_%02d", (Math.random() * 15).toInt())
            val resource = context.resources.getIdentifier(name, "drawable", context.packageName)
            list.add(ImageItem(resource, name))
        }
        loadImageCallback?.onLoadImages(list)
    }
}